tests/components/FilterSearch.test.tsx

```diff
@@ -0,0 +1,298 @@
+      expect(thirdAutocompleteSuggestion).toBeDefined();
+    });
+  });
+
+  it('fills the search bar with an autocomplete result when a user selects it', async () => {
+    render (<FilterSearch searchFields={searchFieldsProp}/>);
+
+    const searchBarElement = screen.getByRole('textbox');
+    userEvent.type(searchBarElement, 'n');
+
+    await waitFor(() => {
+      const firstAutocompleteSuggestion = screen.getByText('first name 1');
+      expect(firstAutocompleteSuggestion).toBeDefined();
+    });
+
+    await waitFor(() => {
+      const secondAutocompleteSuggestion = screen.getByText('first name 2');
+      expect(secondAutocompleteSuggestion).toBeDefined();
+    });
+
+    userEvent.type(searchBarElement, '{arrowdown}');
+    expect(searchBarElement).toHaveValue('first name 1');
+
+    userEvent.type(searchBarElement, '{arrowdown}');
+    expect(searchBarElement).toHaveValue('first name 2');
+  });
+
+  it('triggers a search on pressing "enter" when an autocomplete result is selected', async () => {
+    render (<FilterSearch searchFields={searchFieldsProp}/>);
+
+    const searchBarElement = screen.getByRole('textbox');
+    userEvent.type(searchBarElement, 'n');
+
+    await waitFor(() => {
+      const firstAutocompleteSuggestion = screen.getByText('first name 1');
+      expect(firstAutocompleteSuggestion).toBeDefined();
+    });
+
+    userEvent.type(searchBarElement, '{arrowdown}');
+    expect(searchBarElement).toHaveValue('first name 1');
+
+    userEvent.type(searchBarElement, '{enter}');
+    const expectedSetFilterOptionParam = {
+      fieldId: 'ce_person',
+      matcher: Matcher.Equals,
+      value: 'first name 1' ,
+      displayName: 'first name 1',
+      selected: true
+    };
+    const expectedSetOffsetParam = 0;
+    expect(setFilterOption).toBeCalledWith(expectedSetFilterOptionParam);
+    expect(setOffset).toBeCalledWith(expectedSetOffsetParam);
+
+    const setFilterOptionCallOrder = setFilterOption.mock.invocationCallOrder[0];
+    const setOffsetCallOrder = setOffset.mock.invocationCallOrder[0];
+    expect(setFilterOptionCallOrder).toBeLessThan(setOffsetCallOrder);
+
+    expect(searchOperations.executeSearch).toBeCalled();
+  });
+
+  it('does not trigger a search on pressing "enter" if no autocomplete result is selected', async () => {
+    render (<FilterSearch searchFields={searchFieldsProp}/>);
+
+    const searchBarElement = screen.getByRole('textbox');
+    userEvent.type(searchBarElement, 'n{enter}');
+
+    expect(setFilterOption).not.toBeCalled();
+    expect(setOffset).not.toBeCalled();
+    expect(searchOperations.executeSearch).not.toBeCalled();
+  });
+
+  it('triggers a search when an autocomplete result is clicked', async () => {
+    render (<FilterSearch searchFields={searchFieldsProp}/>);
+
+    const searchBarElement = screen.getByRole('textbox');
+    userEvent.type(searchBarElement, 'n');
+
+    let firstAutocompleteSuggestion;
+
+    await waitFor(() => {
```

Comment:
Without the waitFors, the get functions can't find the elements that I'm looking for.  I assume it's because when you type in a letter, executeFilterSearch is called and can't return and render the autocomplete suggestions before the get function is called.